fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -363,6 +421,27 @@ def add_text(
     for i in range(num_sides):
         if "deluxe" in states[i].model_name:
             hint_msg = SLOW_MODEL_MSG
+    
+    if file_extension == ".pdf":
+        document_text = llama_parse(files[0])
+        post_processed_text = f"""
+        The following is the content of a document:
+
+        {document_text}
+
+        Based on this document, answer the following question:
+
+        {text}
+        """
+    
+    post_processed_text = wrap_query_context(text, post_processed_text)
+
+    text = text[:BLIND_MODE_INPUT_CHAR_LEN_LIMIT]  # Hard cut-off
```

Comment:
We should probably avoid cutting off inputs when dealing with pdf.

Reply:
Fixed. I will only cut off input for images.